src/main/java/hudson/plugins/tfs/browsers/TeamSystemWebAccessBrowser.java

```diff
@@ -47,44 +47,49 @@ private String getBaseUrlString(ChangeSet changeSet) throws MalformedURLExceptio
         if (url != null) {
             baseUrl = DescriptorImpl.getBaseUrl(url);
         } else {
-            baseUrl = String.format("%s/", getServerConfiguration(changeSet)); 
+            String scmUrl = getServerConfiguration(changeSet);
+            if (scmUrl.endsWith("/")) {
+              baseUrl = scmUrl;
+            } else {
+              baseUrl = String.format("%s/", scmUrl);
+            }

```

Comment:
This "ensureTrailingSlash" logic is duplicated in `DescriptorImpl#getBaseUrl` below. Consider calling that method or extracting a helper that both sites call.